packages/runtime/datastore-definitions/src/jsonable.ts

```diff
@@ -3,19 +3,50 @@
  * Licensed under the MIT License.
  */
 
+/**
+ * Type constraint for types that are serializable as JSON or have a custom
+ * alternate type.
+ *
+ * @remarks
+ * Use JsonableTypeWith<never> for just JSON serializable types.
+ */
+export type JsonableTypeWith<T> =
+	| undefined
+	| null
+	| boolean
+	| number
+	| string
+	| T
+	| Internal_InterfaceOfJsonableTypesWith<T>
+	| ArrayLike<JsonableTypeWith<T>>;
+
+/**
+ * @remarks
```

Comment:
Nit, since this is internal dev facing: 
```suggestion
 * @privateRemarks
```

Reply:
We may also want to add a public facing note that this is a temporary hack not intended for public consumption.

Reply:
Is public facing note `@remarks` or something else?

Reply:
Yeah, `@remarks` is generally reserved for user-facing notes that are not part of the short semantic summary.